src/extension/NUnit2FrameworkDriver.cs

```diff
@@ -137,14 +137,32 @@ public string Explore(string filter)
             if (Runner.Test == null)
                 return String.Format(LOAD_RESULT_FORMAT, TestID, _name, _fullname, "Error loading test");
 
-            return Runner.Test.ToXml(true).OuterXml;
+            // Make a copy so that we can edit the list of tests to filter out.
+            var runnerTest = Runner.Test;
```

Comment:
Oh, good catch. The comment says "make a copy" but no copy is made, so something needs to change.

I would say this code needs to be changed to actually make a copy. Otherwise every call to `Explore` with various filters will lose more and more tests. In NUnit 3 framework, I think you can clone a tree of tests while applying a filter at the same time by using a copy constructor that takes a filter. Does this code have something available to it like that?

Reply:
Good observation about the serial use of Explore. I hadn't even thought of that!

There's no such constructor. The pruning method would have to do it recursively, creating new nodes at each level. I think that's simplest here.